File: task-03/Ruby/code4.rb
```ruby
def print_diamond(n)
  lines = []
  (0...n).each do |i|
    lines << ' ' * (n - i - 1) + '*' * (2 * i + 1)
  end
  (n - 2).downto(0) do |i|
    lines << ' ' * (n - i - 1) + '*' * (2 * i + 1)
  end
  lines.join("\n")
end

n = File.read('input.txt').to_i
diamond = print_diamond(n)
File.write('output.txt', diamond)
```

File: task-03/Python/code4.py
```python
def print_diamond(n):
    lines = []
    for i in range(n):
        lines.append(' ' * (n - i - 1) + '*' * (2 * i + 1))
    for i in range(n - 2, -1, -1):
        lines.append(' ' * (n - i - 1) + '*' * (2 * i + 1))
    return '\n'.join(lines)

with open('input.txt', 'r') as infile:
    n = int(infile.read().strip())

diamond = print_diamond(n)

with open('output.txt', 'w') as outfile:
    outfile.write(diamond)
```